src/globals.test.js

```diff
@@ -19,6 +19,68 @@ describe('globals', () => {
     expect(globals.SpansContainer.getSpans()).toEqual([span2]);
   });
 
+  test('GlobalTimer - simple flow', done => {
+    globals.GlobalTimer.setGlobalTimeout(() => {
+      done();
+    }, 1);
+  });
+
+  test('GlobalTimer - override timers', done => {
+    const arr = [];
+    globals.GlobalTimer.setGlobalTimeout(() => {
+      arr.push(1);
+    }, 50);
+    globals.GlobalTimer.setGlobalTimeout(() => {
+      arr.push(2);
+      expect(arr).toEqual([2]);
+      done();
+    }, 100);
+  });
+
+  test('GlobalTimer - clear', done => {
+    const arr = [];
+    globals.GlobalTimer.setGlobalTimeout(() => {
+      arr.push(1);
+    }, 1);
+    globals.GlobalTimer.clearTimer();
+
+    setTimeout(() => {
+      expect(arr).toEqual([]);
+      done();
+    }, 50);
+  });
+
+  test('GlobalTimer - async func flow', done => {
+    const array = [];
+
+    const addToArrayAsync = ms =>
+      new Promise(resolve =>
+        setTimeout(() => {
+          array.push(1);
+          resolve();
+        }, ms)
+      );
+
+    globals.GlobalTimer.setGlobalTimeout(async () => {
+      await addToArrayAsync(1);
+      expect(array).toEqual([1]);
+      done();
+    }, 1);
+  });
+
```

Comment:
maybe worth adding a test for async clear?

Reply:
💪 💪 Adding